openrouteservice/src/main/java/org/heigit/ors/routing/graphhopper/extensions/core/PrepareCore.java

```diff
@@ -229,7 +245,7 @@ void contractNodes() {
                 sortedNodes.clear();
                 int len = prepareGraph.getNodes();
                 for (int node = 0; node < len; node++) {
-                    if (prepareGraph.getLevel(node) != maxLevel)
+                    if (prepareGraph.getLevel(node) < maxLevel)
```

Comment:
`!isCoreNode(node)`?

Reply:
Thanks for the suggestion! Before starting the algorithm all nodes are set to `maxLevel` and assigned a lower level only when contracted. So even though a lower level effectively means that the node won't be included in the core, in this particular example what matters more is whether the node has been already contracted or not. I'm afraid that using a function call `!isCoreNode` for this instead of a direct comparison might make the code more difficult to read and understand.